lib/puppet/transaction/persistence.rb

```diff
@@ -22,14 +22,9 @@ def data
   # @param [String] param_name name of the parameter
   # @return [Object,nil] the system_value
   def get_system_value(resource_name, param_name)
-    if !@old_data["resources"].nil? &&
-       !@old_data["resources"][resource_name].nil? &&
-       !@old_data["resources"][resource_name]["parameters"].nil? &&
-       !@old_data["resources"][resource_name]["parameters"][param_name].nil?
-      @old_data["resources"][resource_name]["parameters"][param_name]["system_value"]
-    else
-      nil
-    end
+    @old_data["resources"][resource_name]["parameters"][param_name]["system_value"]
+  rescue 
+    nil
```

Comment:
Not a huge fan of using an exception for flow control due to performance concerns.